File: testdata/vulnerable.js
// THREAT-001: Spoofing risk — hardcoded credentials.
const password = "1234";
const bypassAuth = true;

function checkAuth(token) {
  // THREAT-001: Hardcoded token comparison.
  if (token === "static-secret") {
    return true;
  }
  return false;
}

// THREAT-002: Tampering risk — no integrity verification.
async function loadConfig() {
  const data = await fetch("https://example.com/config");
  const config = JSON.parse(await data.text());
  return config;
}

// THREAT-003: Repudiation risk — security actions without logging.
async function deleteUser(userId) {
  // No audit log for user deletion.
  await db.users.delete(userId);
}

async function handleLogin(credentials) {
  // No logging of login attempts.
  return await auth.verify(credentials);
}

// THREAT-004: Information disclosure — leaking sensitive data.
function errorHandler(err, req, res, next) {
  console.log("Error with apiKey:", process.env.API_KEY);
  res.json({ error: err.stack });
}

// THREAT-005: Elevation of privilege — privilege escalation.
const child_process = require('child_process');
function runAsRoot() {
  child_process.exec("sudo apt-get install something");
  const role = "superadmin";
}

File: testdata/vulnerable.py
import os
import subprocess
import traceback
import requests
import json

# THREAT-001: Spoofing risk — hardcoded credentials.
password = "test"
bypass_auth = True

def check_auth(token):
    # THREAT-001: Weak auth with hardcoded comparison.
    if token == "static-secret-token":
        return True
    return False

# THREAT-002: Tampering risk — no integrity check.
def fetch_remote_data():
    response = requests.get("https://example.com/api/data")
    data = json.loads(response.text)
    return data

# THREAT-003: Repudiation risk — security actions without audit trail.
def delete_user(user_id):
    # No audit log for destructive operation.
    db.users.delete(user_id)

def handle_login(username, password_input):
    # No logging of login attempts.
    return auth.verify(username, password_input)

# THREAT-004: Information disclosure — exposing stack traces.
def handle_error(request, error):
    traceback.print_exc()
    VERBOSE = True
    return str(e)

# THREAT-005: Elevation of privilege — privilege escalation.
def escalate():
    os.setuid(0)
    subprocess.call(["sudo", "chmod", "777", "/etc/passwd"])
    role = "superadmin"
